rockchip: add Firefly ROC-RK3328-CC support

1715173329 opened the pull request:
Hardware
--------
RockChip RK3328 ARM64 (4 cores)
4GB DDR4 RAM
1x 1000 Base-T
2 LEDs (POWER / USER)
HDMI 2.0
3.5mm TRRS AV jack
Micro-SD slot
eMMC slot
1x USB 3.0 Port
2x USB 2.0 Port

Installation
------------
Uncompress the OpenWrt sysupgrade and write it to a micro SD card using
dd.

1715173329 committed:
uboot-rockchip: add ROC-RK3328-CC support

Add support for the Firefly ROC-RK3328-CC.
Manually generated of-platdata files to avoid swig dependency.

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>

1715173329 committed:
rockchip: add Firefly ROC-RK3328-CC support

Hardware
--------
RockChip RK3328 ARM64 (4 cores)
4GB DDR4 RAM
1x 1000 Base-T
2 LEDs (POWER / USER)
HDMI 2.0
3.5mm TRRS AV jack
Micro-SD slot
eMMC slot
1x USB 3.0 Port
2x USB 2.0 Port

Installation
------------
Uncompress the OpenWrt sysupgrade and write it to a micro SD card using
dd.

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>

1715173329 commented:
ping @aparcar @hauke

hauke commented on package/boot/uboot-rockchip/src/of-platdata/roc-cc-rk3328/dt-decl.h:
```diff
@@ -0,0 +1,24 @@
+/*
+ * DO NOT MODIFY
```
Are these files automatically generated to prevent the inclusion of swig?
If so please mention this in the commit message.

1715173329 replied:
> Are these files automatically generated to prevent the inclusion of swig?

yes

hauke commented on target/linux/rockchip/patches-5.15/105-nanopi-r4s-sd-signalling.patch:
```diff
@@ -10,6 +10,16 @@ userspace or following a kernel panic is always working.
+ 	sd-uhs-sdr12;
+ 	sd-uhs-sdr25;
+ 	sd-uhs-sdr50;
+-	sd-uhs-sdr104;
```
Why do you remove this line?
Please explain in the commit message why you do these changes to the dts files.

1715173329 replied:
the rockchip boards have a common bug when operating uhs speed, which will hang the system during a soft reboot.

hauke replied:
Please add this into a separate commit with the explanation you gave in this comment in the commit comment.

1715173329 replied:
done ;)

1715173329 committed:
rockchip: enable rk805 pinctrl driver

This is required for ROC-RK3328-CC to light up the LED-s which are
connected to rk805's gpio.

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>

1715173329 committed:
rockchip: disable UHS speed for ROC-RK3328-CC

The previously stated speed of sdr-104 in is too high for the hardware
to reliably communicate with some fast SD cards.
Rockchip boards have a common bug when operating uhs speed, which will
hang the system during a soft reboot.

To be on the safe side, lower the speed to workaround.

Signed-off-by: Tianling Shen <cnsztl@immortalwrt.org>

1715173329 commented:
Kindly ping @hauke

hauke commented:
Thank you for the pull request, I merged it.
